### DISCUSSION TITLE
How to query in a recursive way using QueryDSL+spring data jpa+gradle+mysql?

### DISCUSSION THREAD

**Discussion by ddxl123** [querydsl/querydsl#3555] [Q&A]
```
@Entity
public class Department {
	private Long id;
	private Long parentId;
}
```
I want to query all of children departments (include children's children and so on) by a department id in a recursive way with QueryDSL (using 5.0.0).

I know that it can be implemented using the withRecursive method of JPASQLQuery.
I read the Maven configuration in the official tutorial, but gradle doesn't know how to configure it, can you provide an example? 
And provide an example of recursive query? thank you!

There is no @OneToMany or @ManyToOne, only parentId.

Related questions: #2150

**Comment by FirePrayer** [comment]
implementation 'com.querydsl:querydsl-core:5.0.0'
    implementation 'com.querydsl:querydsl-sql:5.0.0'
    implementation 'com.querydsl:querydsl-jpa:5.0.0:jakarta'
    annotationProcessor 'com.querydsl:querydsl-apt:5.0.0:jakarta'
    annotationProcessor 'org.springframework.boot:spring-boot-starter-data-jpa'
